hazelcast/src/main/java/com/hazelcast/map/impl/query/MapQueryEngineImpl.java

```diff
@@ -232,25 +231,22 @@ private IterationType getRetrievalIterationType(Predicate predicate, IterationTy
         return retrievalIterationType;
     }
 
-    private List<Integer> getLocalPartitionIds() {
-        return partitionService.getMemberPartitions(nodeEngine.getThisAddress());
-    }
-
-    private Set<Integer> getAllPartitionIds() {
+    private BitSet getLocalPartitionIds() {
         int partitionCount = partitionService.getPartitionCount();
-        return createSetWithPopulatedPartitionIds(partitionCount);
+        BitSet partitionIds = new BitSet(partitionCount);
+        BitSetUtils.setBits(partitionIds, partitionService.getMemberPartitions(nodeEngine.getThisAddress()));
+        return partitionIds;
     }
 
-    private boolean isResultFromAnyPartitionMissing(Collection<Integer> partitionIds) {
```

Comment:
Afaik this is just a nice to have litter reduction; but it has not relevant for the actual optimization. Correct?

Reply:
Yes, litter reduction, and speedup if you run a local query on a local node only, if no communication implied it speeds up a query by 30-40% 
That's the reason I added two separate commits. It's better do review them separately.